extensions/wikia/Search/classes/Result/ResultHelper.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Wikia\Search\Result;
+
+use Wikia\Search\MediaWikiService, Wikia\Search\Utilities, CommunityDataService, ImagesService, WikiaSearchController, PromoImage;
+
+class ResultHelper
+{
+	const MAX_WORD_COUNT_EXACT_MATCH = 16;
+	const MAX_WORD_COUNT_XWIKI_RESULT = 60;
+
+	/**
+	 * +     * Extends search result with additional data from outside search index, like description and image
+	 * +     *
+	 * +     * @param $result
+	 * +     * @param $pos
+	 * +     * @param $descWordLimit
+	 * +     * @return array
+	 * +     */
+	public static function extendResult($result, $pos, $descWordLimit)
+	{
+
+		$thumbTracking = 'class="wiki-thumb-tracking" data-pos="' . $pos . '" data-event="search_click_wiki-thumb"';
+
+		$commData = new CommunityDataService($result['id']);
+		$imageURL = ImagesService::getImageSrc(
+			$result['id'],
+			$commData->getCommunityImageId(),
+			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
+			WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT)['src'];
+
+		//Fallback: if Curated Mainpage is inaccessible, try to use Special:Promote
+		//TODO: Remove after DAT-3642 is done
+		if (empty($imageURL)) {
+			$imageFileName = PromoImage::fromPathname($result['image_s'])->ensureCityIdIsSet($result['id'])->getPathname();
+			$imageURL = ImagesService::getImageSrcByTitle(
+				(new \CityVisualization)->getTargetWikiId($result['lang_s']),
+				$imageFileName,
+				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_WIDTH,
+				WikiaSearchController::CROSS_WIKI_PROMO_THUMBNAIL_HEIGHT);
+		}//TODO: end
+
+		if (empty($imageURL)) {
+			// display placeholder image if no thumbnail
+			$imageURL = \F::app()->wg->ExtensionsPath . '/wikia/Search/images/wiki_image_placeholder.png';
+			$thumbTracking = 'class="wiki-thumb-tracking" data-pos="' . $pos . '" data-event="search_click_wiki-no-thumb"';
```

Comment:
Is it possible (reasonable) to place those classes and data attributes in template?